src/sharding/Shard.js

```diff
@@ -117,11 +131,23 @@ class Shard extends EventEmitter {
           execArgv: this.execArgv,
         })
         .on('message', this._handleMessage.bind(this))
-        .on('exit', this._exitListener);
+        .on('exit', this._exitListener)
+        .on('disconnect', this._handleDisconnect.bind(this, { parentProcessDisconnect: true }));
     } else if (this.manager.mode === 'worker') {
       this.worker = new Worker(path.resolve(this.manager.file), { workerData: this.env })
         .on('message', this._handleMessage.bind(this))
         .on('exit', this._exitListener);
+
+      this.parentPort = require('worker_threads').parentPort;
+
+      if (this.parentPort) {
+        this.manager.client.setInterval(() => {
+          this.parentPort = require('worker_threads').parentPort;
+          if (!this.parentPort) {
+            this._handleDisconnect({ parentThreadDisconnect: true });
+          }
+        }, this.timeout);
```

Comment:
```suggestion
        const interval = this.manager.client.setInterval(() => {
          this.parentPort = require('worker_threads').parentPort;
          if (!this.parentPort) {
            this._handleDisconnect({ parentThreadDisconnect: true });
            this.manager.client.clearInterval(interval);
          }
        }, this.timeout);
```
interval should be cleared

Reply:
Clearing the interval would be redundant, as the line above will shut down the thread/process regardless of state.

Reply:
the interval isnt set on the actual child process is it? it looks as if its being set on the main process?

Reply:
seems that way, could be why previous attempts at testing the PR went 👎 , I'll see about changing this.